core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -162,6 +162,7 @@ class Partition(val topic: String,
    */
   def makeLeader(controllerId: Int,
                  partitionStateInfo: PartitionStateInfo, correlationId: Int): Boolean = {
+    var leaderHWIncremented = false
```

Comment:
Yeah, I don't mind this pattern, but it's a little awkward to have the block produce a value which is actually just a side effect of the function. Usually I would use that pattern when the entire goal of the block was to produce the right-side value of an assignment statement. I think either approach is bound to be a little unsatisfying.